src/graph/backend/dnnl/patterns/mlp.cpp

```diff
@@ -113,6 +113,99 @@ DNNL_BACKEND_REGISTER_PATTERN_MATCHER_PASS(dnnl, gated_mlp_v1)
             return std::make_shared<larger_partition_kernel_t>();
         });
 
+/*
+//        |          |
+//       deq        deq
+//  \   /        \  /
+//  matmul (gt)  matmul (up)
+//     |          |
+//    unary*      |
+//        \      /   deq
+//         binary   /
+//           |     /
+//           matmul (down)
+//              |
+*/
+DNNL_BACKEND_REGISTER_PATTERN_MATCHER_PASS(dnnl, quantized_gated_mlp)
+        .set_priority(22.1f)
+        .set_kind(partition_kind_t::matmul_post_ops)
+        .set_attr<FCreatePattern>("FCreatePattern",
+                [](const std::shared_ptr<pb_graph_t> &pgraph) -> void {
+                    pm::pb_op_t *deq_up = pgraph->append_op(
+                            graph::op_kind::DynamicDequantize);
+                    pm::pb_op_t *deq_gate = pgraph->append_op(
+                            graph::op_kind::DynamicDequantize);
```

Comment:
Should those be restricted to that same `in_edge`?

Reply:
Hi Mourad, the dequantize ops here are for weights which are different for the two matmul ops (up and gate). But I guess you meant the src should be same for the two matmul ops. Currently, the pattern and the primitive based implementation support different src inputs for the two matmul ops. Do you think we'd better add a restriction for that?

Reply:
Thanks Tao. Yes I meant the src. IIRC, the gated MLP pattern always have shared src, correct? 
I guess the question is hence the actual pattern definition: if optimizations can leverage that src is shared by both matmul (e.g. by keeping reordered buffer in SLM on GPU), it would be safer to define the pattern as such IMHO (and reflect that in tests as well).

Reply:
Thanks. The restriction is added for the patterns now.